content/concepts/payment-system-basics/transaction-basics/finality-of-results.md

```diff
@@ -15,3 +15,41 @@ For any other result code, it can be difficult to determine if the result is fin
 | `tefMAX_LEDGER` | Final when a validated ledger has a sequence number higher than the transaction's `LastLedgerSequence` field, and no validated ledger includes the transaction |
 
 Any other transaction result is potentially not final. In that case, the transaction could still succeed or fail later, especially if conditions change such that the transaction is no longer prevented from applying. For example, trying to send a non-XRP currency to an account that does not exist yet would fail, but it could succeed if another transaction sends enough XRP to create the destination account. A server might even store a temporarily-failed, signed transaction and then successfully apply it later without asking first.
+
+## How can non-final results change?
+
+When you initially submit a transaction, the `rippled` server tentatively applies that transaction to its current open ledger, then returns the tentative [transaction results](transaction-results.html) from doing so. However, the transaction's final result may be very different than its tentative results, for several reasons:
+
+- The transaction may be delayed until a later ledger version, or may never be included in a validated ledger. For the most part, the XRP Ledger follows a principle that all valid transactions should be processed as soon as possible. However, there are exceptions, including:
+
+    - If a proposed transaction has not been relayed a majority of validators by the time a [consensus round](consensus.html) begins, it may be postponed until the next ledger version, to give the remaining validators time to fetch the transaction and confirm that it is valid.
+
+    - If an address sends two different transactions using the same sequence number, at most one of those transactions can become validated. If those transactions are relayed through the network in different paths, a tentatively-successful transaction that some servers saw first may end up failing because the other, conflicting transaction reached a majority of servers first.
+
+    - To protect the network from spam, all transactions must destroy a [transaction cost](transaction-cost.html) in XRP to be relayed throughout the XRP Ledger peer-to-peer network. If heavy load on the peer-to-peer network causes the transaction cost to increase, a transaction that tentatively succeeded may not get relayed to enough servers to achieve a consensus, or may be [queued](transaction-queue.html) for later.
```

Comment:
Just curious about causes the transaction to not be relayed to enough servers or be queued for later -- is it the heavy load on the network? Or is it that the transaction cost increased and the account wasn't able to cover the cost? Or both?

Reply:
Both, because they're related. The load on the network dictates—through a somewhat complex decentralized mechanism—how big of a transaction cost is necessary for a transaction to be relayed throughout the network and proposed for inclusion in a validated ledger.

Every transaction has a fixed cost hardcoded in the `Fee` field when it's signed. Whenever a server sees a new proposed transaction, if the transaction's signed cost isn't high enough for the current load level on that server, the server "drops the transaction on the floor", which is to say it doesn't relay the transaction to its peers nor does it apply the transaction to its open ledger or include it in any proposals for validation. So if enough servers _aren't_ under heavy load that the transaction gets passed around to a majority of validators who aren't busy, then there's a good chance the transaction may become validated anyway. Load among servers in the network tends to be correlated (they're processing the same transactions, after all) so the transaction might not make it to enough servers. Also, the topology of the peer-to-peer network can affect this, since even if most of the servers aren't busy, if you submit to a server that has to relay things through busy peers to get to the rest of the network, then the necessary cost is going to vary a bit there.

There's also the open ledger cost, which works a little differently since transactions that don't make the cut are queued & still relayed to peers if they're "likely to succeed".

By the way, it matters less how much the account _could_ pay, and more how much the transaction "promises" to pay with that hardcoded, signed transaction cost. If a transaction is signed with a cost of, say, 12 drops, then if it gets relayed through the network and included in a validated ledger, _it will definitely destroy those 12 drops of XRP_ if at all possible. The only exceptions are if a transaction promises to pay more than its sender has. If the servers notice that while looking at the transaction provisionally, they'll throw it out with a `terINSUF_FEE_B` code. If for some reason the transaction makes it to enough servers to reach a consensus and _then_ it turns out it doesn't have enough XRP (probably something else changed how much XRP the account had at the last minute) then it consumes as much as possible and fails with a tec code. (Not sure which one, maybe `tecUNFUNDED`?)